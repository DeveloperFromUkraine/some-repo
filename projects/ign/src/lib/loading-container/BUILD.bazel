load("@angular//:index.bzl", "ng_module")
load("@io_bazel_rules_sass//:defs.bzl", "sass_binary")

package(default_visibility = ["//visibility:public"])

sass_binary(
    name = "style",
    src = "loading-container.scss",
)

ng_module(
    name = "loading-container-module",
    srcs = [
        "loading-container.module.ts",
        "loading-container.component.ts",
    ],
    assets = [":loading-container.css", "loading-container.html"],
    tsconfig = "//:tsconfig.json",
    deps = [
        "@npm//tslib",
        "@npm//@types",
        "@angular//packages/core",
        "@angular//packages/common",
        "@angular_material//src/lib:material",
        "//projects/ign/src/lib/localization:localization-translation-module"
    ]
)
