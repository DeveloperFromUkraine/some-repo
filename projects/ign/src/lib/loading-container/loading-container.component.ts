import { Component, Input, OnInit, ChangeDetectorRef } from '@angular/core';

import { TranslationPipe } from '../localization/translation.pipe';

const DEFAULT_LOADING_DELAY_MS = 400;

@Component({
  selector: 'ign-loading-container',
  styleUrls: ['./loading-container.css'],
  templateUrl: './loading-container.html',
})
export class LoadingContainerComponent implements OnInit {
  @Input() loading: boolean;
  @Input() ariaLabel: string;
  spinnerAriaLabel: string;
  pastDelay: boolean;
  private delay: number;
  translatePipe: TranslationPipe;

  constructor(changeRef: ChangeDetectorRef) {
    this.delay = DEFAULT_LOADING_DELAY_MS;
    this.loading = true;
    this.pastDelay = false;
    this.translatePipe = new TranslationPipe(changeRef);
    this.spinnerAriaLabel = this.translatePipe.transform('LOADING');
  }

  ngOnInit() {
    if (this.ariaLabel) {
      this.spinnerAriaLabel = this.ariaLabel;
    }
    setTimeout(() => {
      this.pastDelay = true;
    }, this.delay);
  }
}
