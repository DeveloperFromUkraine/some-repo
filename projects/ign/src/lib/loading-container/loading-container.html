<div [hidden]="loading">
    <ng-content></ng-content>
  </div>
  <div *ngIf="loading" attr.aria-label="{{spinnerAriaLabel | translateIDS}}" role="alert" tabindex="0">
    <div *ngIf="pastDelay" class="spinner-container">
      <mat-spinner></mat-spinner>
    </div>
</div>