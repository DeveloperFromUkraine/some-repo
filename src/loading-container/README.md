# Loading Container

## Description

Loading container is a wrapper (higher level) component around your content. It will show a spinner after a period (right now 400ms) until the consumer declares the content loaded. Most developers needing to consume the loading container will simply wrap their existing on content in an `ign-loading-container` and let the loading container know when the content is loaded.

## API
  ### Selector
    * ign-loading-container

  ### Exported As
    * LoadingContainerComponent
    
  ### Properties
    * '@Input' loading (input): boolean
      *  Loading status, defaults to true.

## Implementation HTML
  <ign-loading-container [loading]="BOOLEAN">
  </ign-loading-container>

## Implementation TS
  None
  
## Usage

The following is a usage example demonstrating the above descriptions:

```
<div *ngIf="(legalEntities$ | async) || []; let items">
  <ign-loading-container [loading]="!loaded">
    <!-- Empty State View -->
    <ign-empty-state *ngIf="items?.length === 0">
      <ign-breadcrumbs></ign-breadcrumbs>
      <ign-empty-state-image></ign-empty-state-image>
      <ign-empty-state-title>{{ 'ACTIVATION.EMPTY_LEGAL_ENTITY' | translate }}</ign-empty-state-title>
      <ign-empty-state-description>
        {{ 'ACTIVATION.LEGAL_ENTITY_EXPLANATION' | translate }}
      </ign-empty-state-description>
      <ign-button-group>
        <button mat-button mat-raised-button color="accent" type="button" [routerLink]="['./add']">
          {{ 'COMMON.ADD' | translate }}
        </button>
      </ign-button-group>
    </ign-empty-state>

    <!-- List Content View -->
    <ign-list-content *ngIf="items?.length !== 0">
      <ign-list-selection>
        <ign-list-selection-item *ngFor="let item of items" [routerLink]="['./', item.id]" [routerLinkActive]="'active'" [key]="item.id"
          [title]="item.name">
        </ign-list-selection-item>
        <ign-list-selection-add-button tabindex="-1" [routerLink]="['./add']"></ign-list-selection-add-button>
      </ign-list-selection>
      <ign-list-content-panel>
        <ign-breadcrumbs></ign-breadcrumbs>

        <!--State for when items exist, but user hasn't selected one yet  -->
        <ign-list-content-unselected-state [hidden]="isLegalEntitySelected">
          Please select a Legal Entity
        </ign-list-content-unselected-state>

        <!--State for inspecting selected item  -->
        <ign-list-content-card [hidden]="!isLegalEntitySelected">
          <router-outlet (activate)="isLegalEntitySelected=true" (deactivate)="isLegalEntitySelected=false"></router-outlet>
        </ign-list-content-card>

      </ign-list-content-panel>
    </ign-list-content>
  </ign-loading-container>
</div>
```
