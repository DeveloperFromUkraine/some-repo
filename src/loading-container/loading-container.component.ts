import {Component, Input, OnInit} from '@angular/core';

const DEFAULT_LOADING_DELAY_MS = 400;

@Component({
  selector: 'ign-loading-container',
  styleUrls: ['./loading-container.scss'],  
  templateUrl: './loading-container.html',
})

export class LoadingContainer implements OnInit {
  @Input() loading: boolean;
  pastDelay: boolean;
  private delay: number;
  private delayTimeout: any;

  constructor() {
    this.delay = DEFAULT_LOADING_DELAY_MS;
    this.loading = true;
    this.pastDelay = false;
  }

  ngOnInit() {
    this.delayTimeout = setTimeout(() => {
      this.pastDelay = true;
    }, this.delay);
  }
}
