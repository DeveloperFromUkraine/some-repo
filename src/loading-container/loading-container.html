<div *ngIf="!loading">
    <ng-content></ng-content>
  </div>
  <div *ngIf="loading" attr.aria-label="{{spinnerArialLabel}}" role="alert">
    <div *ngIf="pastDelay" class="spinner-container">
      <mat-spinner></mat-spinner>
    </div>
</div>