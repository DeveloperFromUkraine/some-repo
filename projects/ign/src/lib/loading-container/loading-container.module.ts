import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { MatProgressSpinnerModule } from '@angular/material';

import { LoadingContainerComponent } from './loading-container.component';
import { TranslationModule } from '../localization/translation.module';

@NgModule({
    imports: [
        CommonModule,
        TranslationModule,
        MatProgressSpinnerModule
    ],
    exports: [
        LoadingContainerComponent
    ],
    declarations: [
        LoadingContainerComponent
    ]
})
export class LoadingContainerModule {

}
