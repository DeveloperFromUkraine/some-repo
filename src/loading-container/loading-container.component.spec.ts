import { ComponentFixture, TestBed } from "@angular/core/testing";
import { LoadingContainerComponent } from "./loading-container.component";
import { ComponentTest } from "../../test/test-bed/component";
import { MatProgressSpinnerModule } from "@angular/material";


describe('LoadingContainer', () => {
    let fixture: ComponentFixture<LoadingContainerComponent>;

    beforeEach(async () => {
        await ComponentTest.createTestBed([MatProgressSpinnerModule], [LoadingContainerComponent]);
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(LoadingContainerComponent);

        fixture.detectChanges();
    }); 

    it('should match snapshot', async () => {
        expect(fixture).toMatchSnapshot();
    });
});