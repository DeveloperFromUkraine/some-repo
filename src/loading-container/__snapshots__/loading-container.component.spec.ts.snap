// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingContainer should match snapshot 1`] = `
<ign-loading-container
  delay={[Function Number]}
  delayTimeout={[Function Number]}
  loading={[Function Boolean]}
  pastDelay="false"
>
  
  <div>
    
    
    
    
  
  </div>
  
</ign-loading-container>
`;
